mods/cnc/maps/gdi09/gdi09.lua

```diff
@@ -0,0 +1,216 @@
+}
+
+AutoAttackWaves = { Nod1, Nod2, Nod3, Nod4, Nod5, Auto1, Auto2, Auto3, Auto4, Auto5, Auto6, Auto7, Auto8 }
+
+StationaryGuards = { Actor174, Actor173, Actor182, Actor183, Actor184, Actor185, Actor186, Actor187 , Actor199, Actor200, Actor201, Actor202, Actor203, Actor204}
+
+StartStationaryGuards = function()
+	Utils.Do(StationaryGuards, function(unit)
+		if not unit.IsDead then
+			unit.Patrol( { unit.Location } , true, 20)
+		end
+	end)
+end
+
+StartWaves = function()
+	SendWaves(1, AutoAttackWaves)
+end
+
+SendWaves = function(counter, Waves)
+	if counter <= #Waves then
+		local team = Waves[counter]
+		SendAttackWave(team)
+		Trigger.AfterDelay(DateTime.Seconds(team.delay), function() SendWaves(counter + 1, Waves) end)
+	end
+end
+
+SendAttackWave = function(team)
+	for type, amount in pairs(team.units) do
+		count = 0
+		local actors = enemy.GetActorsByType(type)
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle and count < amount then
+				SetAttackWaypoints(actor, team.waypoints)
+				IdleHunt(actor)
+				count = count + 1
+			end
+		end)
+	end
+end
+
+SetAttackWaypoints = function(actor, waypoints)
+	if not actor.IsDead then
+		Utils.Do(waypoints, function(waypoint)
+			actor.AttackMove(waypoint.Location)
+		end)
+	end
+end
+
+StartPatrols = function()
+	for k, team in pairs(Patrols) do
+		local group = 1
+		for type, amount in pairs(team.units) do
+			for i=1,amount,1 do
+				Reinforcements.Reinforce(enemy, { type }, { team.waypoints[team.initialWaypointPlacement[group]] }, 0, function(unit)
+					ReplenishPatrolUnit(unit, handofnod, team.waypoints, team.wait)
+				end)
+			end
+			group = group + 1
+		end
+	end
+	Patrols = nil
+end
+
+ReplenishPatrolUnit = function(unit, building, waypoints, waitatwaypoint)
+	unit.Patrol(waypoints, true, waitatwaypoint)
+	Trigger.OnKilled(unit, function()
+		local queueUnit = { unit = { unit.Type }, atbuilding = { building }, waypoints = waypoints }
+		PatrolProductionQueue[#PatrolProductionQueue + 1] = queueUnit
+	end)
+end
+
+SendGDIReinforcements = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Reinforcements.ReinforceWithTransport(player, InsertionHelicopterType, GDIHeliReinfUnits, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location }, { GDIHeliLZ.Location + CVec.New(20, 0) })
+end
+
+SendGDIReinforcementChinook = function()
+	Reinforcements.ReinforceWithTransport(player, 'tran', nil, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location })
+end
+
+MarkCompletedObjectiveBunkers = function()
+	if not player.IsObjectiveCompleted(gdiMainObjectiveBunkers) then
+		player.MarkCompletedObjective(gdiMainObjectiveBunkers)
+	end
+end
+
+SpawnGunboat = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Actor.Create("boat", true, { Owner = player, Facing = 0, Location = waypoint23.Location })
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("GDI")
+	enemy = Player.GetPlayer("Nod")
```

Comment:
It would be good to name those GDI and Nod respectively, but this can be left to when we port the mission to the global script.